database/service/clean.go

```diff
@@ -0,0 +1,31 @@
+// Copyright (c) 2023 Target Brands, Inc. All rights reserved.
+//
+// Use of this source code is governed by the LICENSE file in this repository.
+
+package service
+
+import (
+	"github.com/go-vela/types/constants"
+	"github.com/go-vela/types/database"
+	"github.com/go-vela/types/library"
+	"github.com/sirupsen/logrus"
+)
+
+// CleanServices updates services to an error with a created timestamp prior to a defined moment.
+func (e *engine) CleanServices(before int64) (int64, error) {
+	logrus.Tracef("cleaning pending or running steps in the database created prior to %d", before)
+
+	s := new(library.Service)
+	s.SetStatus(constants.StatusError)
```

Comment:
For service and step do we want to also add the admin's error message?

And should we set the finished time?

```suggestion
	s := new(library.Service)
	s.SetStatus(constants.StatusError)
	s.SetError(msg)
	s.SetFinished(time.Now().UTC().Unix())
```